### PR TITLE
LPS-204012 first play with playwright

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:playwright

**Comment by marcogalluzzi** [comment]
Do the tests works? If you enable the FF the Publish button shows 2 dropdown options on click.

**Comment by AliciaGarciaGarcia** [comment]
> Do the tests works? If you enable the FF the Publish button shows 2 dropdown options on click.

No, thats my point with this draft

1 if works the login (i think that no)
2 how the errors are shown :)

**Comment by AliciaGarciaGarcia** [comment]
ci:test:playwright

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:playwright

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:playwright

File: modules/test/playwright/fixtures/knowledgeBase/knowldegbeBase.fixure.ts
Comment (austinchiang) [inline modules/test/playwright/fixtures/knowledgeBase/knowldegbeBase.fixure.ts:1]: knowledgeBase.fixure

Comment (AliciaGarciaGarcia) [reply modules/test/playwright/fixtures/knowledgeBase/knowldegbeBase.fixure.ts:1]: :D
Thanks!!!

**Comment by AliciaGarciaGarcia** [comment]
ci:test:playwright

**Comment by AliciaGarciaGarcia** [comment]
ci:test:playwright

**Comment by boton** [comment]
ci:test:playwright

**Comment by boton** [comment]
ci:test:playwright

**Comment by AliciaGarciaGarcia** [comment]
I will rebase and resend

**Comment by boton** [comment]
ci:test:playwright

**Comment by boton** [comment]
ci:test:playwright

File: modules/test/playwright/pages/knowledge-base-web/KnowledgeBaseViewArticle.page.ts [hunk 1]
```diff
@@ -11,13 +11,14 @@ export class KnowledgeBaseViewArticlePage {
...
+		this.showActionsButton = page
+			.getByLabel('Show Actions')
+			.and(page.locator('[aria-haspopup]'));
 		this.page = page;
 	}
 
 	async deleteKnowledgeBaseArticle() {
-		// await this.showActionsButton.locator('[aria-haspopup]').waitFor();
-
+		await this.showActionsButton.waitFor();
```
Comment (marcogalluzzi) [inline modules/test/playwright/pages/knowledge-base-web/KnowledgeBaseViewArticle.page.ts:21]: Would it be a good idea to put the `waitFor` inside the `constructor` so we know `showActionsButton` is already available when using it? Also, can the `waitFor` + second condition be replaced more clearly by an [`expect`](https://playwright.dev/docs/actionability#assertions) ?

Comment (boton) [reply modules/test/playwright/pages/knowledge-base-web/KnowledgeBaseViewArticle.page.ts:21]: Nop, the constructor code must be sync

Comment (AliciaGarciaGarcia) [reply modules/test/playwright/pages/knowledge-base-web/KnowledgeBaseViewArticle.page.ts:21]: the async deleteKnowledgeBaseArticle() {
is not a test, so an expect is... not expected

**Comment by AliciaGarciaGarcia** [comment]
fail all the test!

image

**Comment by marcogalluzzi** [comment]
All errors are because of this:

```
Error: locator.click: Page closed
=========================== logs ===========================
waiting for getByRole('menuitem', { name: 'Basic Article' })
============================================================

   at ../pages/knowledge-base-web/KnowledgeBaseFoldersAndArticles.page.ts:55

  53 | 		await this.goToFoldersAndArticles();
  54 | 		await this.newButton.click();
> 55 | 		await this.basicArticleMenuItem.click();
     | 		                                ^
  56 | 	}
  57 |
  58 | 	async deleteKnowledgeBaseArticle(page: Page, title: string) {

    at KnowledgeBaseFoldersAndArticlesPage.goToCreateNewArticle (/opt/dev/projects/github/liferay-portal/modules/test/playwright/pages/knowledge-base-web/KnowledgeBaseFoldersAndArticles.page.ts:55:35)
    at KnowledgeBaseFoldersAndArticlesPage.publishNewKnowledgeBaseArticle (/opt/dev/projects/github/liferay-portal/modules/test/playwright/pages/knowledge-base-web/KnowledgeBaseFoldersAndArticles.page.ts:34:3)
    at KnowledgeBaseHelper.createNewKnowledgeBaseArticle (/opt/dev/projects/github/liferay-portal/modules/test/playwright/helpers/knowledgeBase/KnowledgeBaseHelper.ts:37:3)
    at /opt/dev/projects/github/liferay-portal/modules/test/playwright/tests/knowledge-base-web/knowledge-base-web.spec.ts:27:2
```

**Comment by AliciaGarciaGarcia** [comment]
ci:test:playwright

**Comment by boton** [comment]
Hi @marcogalluzzi you are right, I'm suffering the same types of errors locally.

image

I suspect that some clicks are failing because are happening before the UI is JS rendered.

I will try to fix it in similar way https://github.com/liferay-lima/liferay-portal/pull/5020/files/9d4047216913d62be620c671cb5e3ea415e3ecac..64cf060c23f4a5b702540b369001f06bdb953db2#diff-c13c81a9641f8fd45f36729499ccb3c11fc598a49410b908c49be724e4a4253cR16

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1307401_8618

**Comment by boton** [comment]
ci:test:playwright

**Comment by boton** [comment]
My work here is almost done! what do you think @AliciaGarciaGarcia?
I need to rebase in master and use the new login thing

image

**Comment by boton** [comment]
ci:test:playwright

**Comment by boton** [comment]
ci:test:playwright

**Comment by boton** [comment]
ci:test:playwright

**Comment by boton** [comment]
ci:test:playwright

**Pull request closed without merging**